src/snowflake/snowpark/_internal/open_telemetry.py

```diff
@@ -56,6 +65,60 @@ def open_telemetry_context_manager(func, dataframe):
         yield
 
 
+@contextmanager
+def open_telemetry_udf_context_manager(func, parameters):
+    # trace when required package is installed
+    if open_telemetry_found:
+        class_name = func.__qualname__
+        name = func.__name__
+        tracer = trace.get_tracer(f"snow.snowpark.stored_procedure:{class_name}")
+
+        with tracer.start_as_current_span(name) as cur_span:
+            try:
+                # first try to get func if it is udf, then try to get handler if it is udtf/udaf, if still None, means it is
+                # loading from file
+                udf_func = (
+                    parameters.get("func")
+                    if parameters.get("func")
+                    else parameters.get("handler")
+                )
+                # if udf_func is not None, meaning it is a function or handler, get handler_name from it, otherwise find
```

Comment:
nit: I think this is what you mean?
```suggestion
                # if udf_func is not None, meaning it is a udf function or udtf handler class, get handler_name from it, otherwise find
```